spec/message.abnf

```diff
@@ -1,43 +1,46 @@
-message = pattern / complex-message
+message = simple-message / complex-message
 
-complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
+simple-message = [simple-start pattern]
+simple-start = simple-start-char / text-escape / expression
+pattern = *(text-char / text-escape / expression)
 
-declaration = input-declaration / local-declaration
+complex-message = *(declaration [s]) complex-body
+declaration = input-declaration / local-declaration / reserved-statement
 input-declaration = input [s] variable-expression
 local-declaration = local s variable [s] "=" [s] expression
 
-body = quoted-pattern
-     / (selectors 1*([s] variant))
-
+complex-body = quoted-pattern
+             / ((selectors / reserved-statement) 1*([s] variant))
 quoted-pattern = "{{" pattern "}}"
-pattern = *(text / expression)
+
 selectors = match 1*([s] expression)
-variant = when 1*(s key) [s] quoted-pattern
+variant = key *(s key) [s] quoted-pattern
 key = literal / "*"
 
 expression = literal-expression / variable-expression / function-expression
 literal-expression = "{" [s] literal [s annotation] [s] "}"
 variable-expression = "{" [s] variable [s annotation] [s] "}"
 function-expression = "{" [s] annotation [s] "}"
-annotation = (function *(s option)) / reserved / private-use
+annotation = (function *(s option))
+           / reserved-annotation
+           / private-use-annotation
 
 literal = quoted / unquoted
 variable = "$" name
 function = (":" / "+" / "-") identifier
 option = identifier [s] "=" [s] (literal / variable)
 
-; reserved keywords are always lowercase
-input = %s"input"
-local = %s"local"
-match = %s"match"
-when  = %s"when"
+input = %s".input"
+local = %s".local"
+match = %s".match"
 
-text = 1*(text-char / text-escape)
-text-char = %x0-5B         ; omit \
-          / %x5D-7A        ; omit {
-          / %x7C           ; omit }
-          / %x7E-D7FF      ; omit surrogates
-          / %xE000-10FFFF
+simple-start-char = %x0-2D         ; omit .
```

Comment:
Nit: it looks like we don't append `-char` to other `-start` productions in our grammar.

```suggestion
simple-start = %x0-2D         ; omit .
```

Reply:
Note that there already is another rule that's called `simple-start`. If necessary, could we take care of any such editorial rule renames in a separate follow-up PR?

Reply:
Ah, I see. I'll open a new PR for this.